lib/unhangout-routes.js

```diff
@@ -788,6 +789,121 @@ module.exports = {
         // Starting and stopping events
         //
 
+        function generateUserData(attrs) {   
+
+            var event = db.events.get(attrs.id);
+            var history = event.get("history");
+
+            var userData = [];
+
+            _.each(history.event, function(elapsedTime, userId) {
+                
+                //Get the user object for a specific ID 
+                var user = db.users.get(userId);
+
+                // Find all userIDs for users who shared sessions with me at this event.
+                var userIds = event.getUserIdsSharingSessionsWith(userId);
+                
+                // get user objects for the IDs.
+                var users = _.map(userIds, function(userId) { 
+                    return db.users.get(userId); 
+                });
+
+                userData.push({ user: user, users: users });
+
+            });
+
+            return userData; 
+
+        }
+
+        app.get("/followup/event/:id/participant_:participantIndex", utils.ensureAuthenticated, utils.ensureSuperuser, function(req, res) {
+
+            //Call generate user data function to grab history of event participants
+            var userData = generateUserData(req.params);   
+            userData.unshift(null);
+
+            if(userData) {
+                res.render("email_preview_template.ejs", {
+                    participantIndex: req.params.participantIndex,
+                    userData: userData,
+                    userDataCount: userData.length - 1,
+                    event: db.events.get(req.params.id)
+                });
+            }
+        });
+
+        app.post("/followup/event/:id/sent/", utils.ensureAuthenticated, utils.ensureSuperuser, function(req, res) {
+
+            //Call generate user data function to grab history of event participants
+            var userData = generateUserData(req.params);
+
+            var mandrill_client = new mandrill.Mandrill('YOUR MANDRILL API KEY');
+
+            var message = {}; 
+            var emailsSent = 0;
+
+            //Execute a loop to send emails
+            _.each(userData, function(user, participantIndex) {
+                
+                res.render("email_template.ejs", {
+                    event: db.events.get(req.params.id),
```

Comment:
Would prefer to pull the event once -- basically, add a ``var event = db.events.get(req.params.id);`` outside the loop, and just reference that event here.  Lookups are super fast now because the db is all in-memory, but we are likely going to some day switch to a real database, and the lookup costs to repeatedly query for the event will become a bother then.